server/requeststojointeamhandler.go

```diff
@@ -0,0 +1,93 @@
+package server
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/fluidkeys/api/datastore"
+	"github.com/fluidkeys/api/v1structs"
+	"github.com/fluidkeys/crypto/openpgp"
+	"github.com/gofrs/uuid"
+	"github.com/gorilla/mux"
+)
+
+func listRequestsToJoinTeamHandler(w http.ResponseWriter, r *http.Request) {
+	apparentSignerKey, err := getAuthorizedUserPublicKey(r)
+	if err == errAuthKeyNotFound {
+		writeJsonError(w,
+			fmt.Errorf("public key that signed the roster has not been uploaded"),
+			http.StatusBadRequest)
+		return
+	} else if err != nil {
+		writeJsonError(w, err, http.StatusBadRequest)
+		return
+	}
+
+	teamUUID, err := uuid.FromString(mux.Vars(r)["teamUUID"])
+	if err != nil {
+		writeJsonError(w, err, http.StatusBadRequest)
+		return
+	}
+
+	dbTeam, err := datastore.GetTeam(nil, teamUUID)
+	if err == datastore.ErrNotFound {
+		writeJsonError(w, fmt.Errorf("team not found"), http.StatusNotFound)
+		return
+
+	} else if err != nil {
+		writeJsonError(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	var keyring openpgp.EntityList = []*openpgp.Entity{&apparentSignerKey.Entity}
```

Comment:
I think this next section would be well worth factoring out into a function called `validateDataSignedByKey(data, signature, apparentlySignedByKey` perhaps with a comment like:

> We check that 1. the roster roster is valid and 2. the signature originated from the specified key. We achieve 2. by passing in a keyring containing *only* the key to test against. Any other key will fail (as the key will be unavailable to check the signature)

Reply:
Actually I propose that I'll merge this then do that for both endpoints separately

Reply:
Double actually, what you want is `checkRosterSignature`, but I should make it clearer what that function is doing

Reply:
Leave this how it is and I'll tackle it in another PR :)